src/main/java/com/stablekernel/rxespresso/RxIdlingResource.java

```diff
@@ -129,12 +133,37 @@ public void onNext(T t) {
 
     private <T> void onFinally(Observable.OnSubscribe<T> onSubscribe, final String finalizeCaller) {
         int activeSubscriptionCount = subscriptions.decrementAndGet();
+        removeOnSubscribe(onSubscribe);
+
         if (LOG_LEVEL >= DEBUG) {
-            Log.d(TAG, onSubscribe + " - " + finalizeCaller + ": " + activeSubscriptionCount);
+            Log.d(TAG, "Subscription finished because: " + finalizeCaller + ". Active subscription count is now: " + activeSubscriptionCount);
+
+            if (LOG_LEVEL >= VERBOSE) {
+                // remaining subscriptions:
+                Log.d(TAG, ">>>>>>> remaining subscriptions ");
+                printStackTraces();
+                Log.d(TAG, "<<<<<<<<");
+            }
```

Comment:
Maybe we could protect the logging by some kind of time window.  Since this relies on Rx anyway, it could be some kind of debounce type operation.